components/provisioner/internal/provisioning/converters.go

```diff
@@ -159,24 +206,56 @@ func clusterConfigToGraphQLConfig(config interface{}) gqlschema.ClusterConfig {
 
 func gardenerConfigToGraphQLConfig(config model.GardenerConfig) gqlschema.ClusterConfig {
 
+	providerSpecificConfig := providerSpecificConfigToGQLConfig(config.ProviderSpecificConfig)
+
 	return gqlschema.GardenerConfig{
-		Name:              &config.Name,
-		ProjectName:       &config.ProjectName,
-		KubernetesVersion: &config.KubernetesVersion,
-		NodeCount:         &config.NodeCount,
-		DiskType:          &config.DiskType,
-		VolumeSize:        &config.VolumeSize,
-		MachineType:       &config.MachineType,
-		TargetProvider:    &config.TargetProvider,
-		TargetSecret:      &config.TargetSecret,
-		Cidr:              &config.Cidr,
-		Region:            &config.Region,
-		Zone:              &config.Zone,
-		AutoScalerMin:     &config.AutoScalerMin,
-		AutoScalerMax:     &config.AutoScalerMax,
-		MaxSurge:          &config.MaxSurge,
-		MaxUnavailable:    &config.MaxUnavailable,
+		Name:                   &config.Name,
+		ProjectName:            &config.ProjectName,
+		KubernetesVersion:      &config.KubernetesVersion,
+		NodeCount:              &config.NodeCount,
+		DiskType:               &config.DiskType,
+		VolumeSizeGb:           &config.VolumeSizeGB,
+		MachineType:            &config.MachineType,
+		Provider:               &config.Provider,
+		Seed:                   &config.Seed,
+		TargetSecret:           &config.TargetSecret,
+		WorkerCidr:             &config.WorkerCidr,
+		Region:                 &config.Region,
+		AutoScalerMin:          &config.AutoScalerMin,
+		AutoScalerMax:          &config.AutoScalerMax,
+		MaxSurge:               &config.MaxSurge,
+		MaxUnavailable:         &config.MaxUnavailable,
+		ProviderSpecificConfig: providerSpecificConfig,
+	}
+}
+func providerSpecificConfigToGQLConfig(config string) gqlschema.ProviderSpecificConfig {
```

Comment:
We decode from json to model structures. We could decode to graphql structures directly.